### PR TITLE
Fix apt_repository uri single/double quotes and spaces

### PR DESCRIPTION
**Pull request by vsingh-msys**
## Description

As apt repository resource appended double/single quotes and this be problematic. also seems we are adding extra spaces.

Here some example of the apt source list

```
deb http://deb.debian.org/debian buster main
deb-src http://deb.debian.org/debian buster main

deb http://deb.debian.org/debian-security/ buster/updates main
deb-src http://deb.debian.org/debian-security/ buster/updates main

deb http://deb.debian.org/debian buster-updates main
deb-src http://deb.debian.org/debian buster-updates main
```
## Related Issue

Fixes https://github.com/chef/chef/issues/6524

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

**Comment by lamont-granquist** [comment]
should you `URI.escape` this to deal with spaces?

**Comment by vsingh-msys** [comment]
Not exactly, here the spaces only mean we are appending while constructing the string in order to make alignment of `deb` & `deb-src` 
eg.
Actual content without spaces
```
deb http://deb.debian.org/debian buster-updates main
deb-src http://deb.debian.org/debian buster-updates main
```

Constructed content:
```
deb      http://deb.debian.org/debian buster-updates main
deb-src  http://deb.debian.org/debian buster-updates main
```
Appending spaces is not a good idea so removed these spaces from

https://github.com/chef/chef/blob/36d1ad5cf9e77559f017db5774a399a2d9626051/lib/chef/provider/apt_repository.rb#L334-L335

**Comment by vsingh-msys** [comment]
And yes for URI validation we can something add like

```
uri = URI.excape(uri)
uri = URI.parse(uri)
```
So that it raise exception if user set a invalid URI?

**Comment by tas50** [comment]
https://ruby-doc.org/stdlib-2.6.5/libdoc/uri/rdoc/URI/Escape.html will let us just take the input value and escape it.

**Comment by lamont-granquist** [comment]
Yeah but were the quotes introduced to handle spaces in the URI:

```
deb      "http://deb.debian.org/debian with spaces" buster-updates main
```

Should that instead be:

```
deb      http://deb.debian.org/debian%20with%20spaces buster-updates main
```

File: lib/chef/provider/apt_repository.rb [hunk 1]
```diff
@@ -331,8 +329,8 @@ def build_repo(uri, distribution, components, trusted, arch, add_src = false)
                    "[" + options.join(" ") + "]"
                  end
         info = [ optstr, uri, distribution, components ].compact.join(" ")
-        repo =  "deb      #{info}\n"
-        repo << "deb-src  #{info}\n" if add_src
+        repo =  "deb #{info}\n"
```
Comment (tas50) [inline lib/chef/provider/apt_repository.rb]: This makes it much easier to read. We should just leave this

Comment (vsingh-msys) [reply lib/chef/provider/apt_repository.rb]: > Yeah but were the quotes introduced to handle spaces in the URI:
> 
> ```
> deb      "http://deb.debian.org/debian with spaces" buster-updates main
> ```
> 
> Should that instead be:
> 
> ```
> deb      http://deb.debian.org/debian%20with%20spaces buster-updates main
> ```

sure @lamont-granquist will do thanks

**Comment by vsingh-msys** [comment]
> Yeah but were the quotes introduced to handle spaces in the URI:
> 
> ```
> deb      "http://deb.debian.org/debian with spaces" buster-updates main
> ```
> 
> Should that instead be:
> 
> ```
> deb      http://deb.debian.org/debian%20with%20spaces buster-updates main
> ```



> Yeah but were the quotes introduced to handle spaces in the URI:
> 
> ```
> deb      "http://deb.debian.org/debian with spaces" buster-updates main
> ```
> 
> Should that instead be:
> 
> ```
> deb      http://deb.debian.org/debian%20with%20spaces buster-updates main
> ```

sure @lamont-granquist will apply changes and will update PR. thanks

**Pull request merged**